src/style.rs

```diff
@@ -23,6 +23,7 @@ pub struct ProgressStyle {
     // how unicode-big each char in progress_chars is
     char_width: usize,
     format_map: HashMap<&'static str, fn(&ProgressState) -> String>,
+    tab_width: usize,
```

Comment:
If we make this a `tab_str: String` we don't have to do `iter::repeat()` thing in `format_state()`. We can still keep `with_tab_width()` taking `usize` and expand it there.